django_fido/views.py

```diff
@@ -1,139 +1,193 @@
-"""Views for U2F registration and login."""
+"""Views for FIDO 2 registration and login."""
 from __future__ import unicode_literals
 
+import base64
 import logging
 from abc import ABCMeta, abstractmethod
-from copy import deepcopy
 from http.client import BAD_REQUEST
-from typing import Optional
-from urllib.parse import urlsplit, urlunsplit
+from typing import Dict, Optional, Tuple
 
 from django.conf import settings
 from django.contrib.auth import authenticate, get_user_model, login
+from django.contrib.auth.base_user import AbstractBaseUser
 from django.contrib.auth.mixins import LoginRequiredMixin
 from django.contrib.auth.views import LoginView
-from django.http import JsonResponse
+from django.forms import Form
+from django.http import HttpRequest, HttpResponse, JsonResponse
 from django.shortcuts import redirect
 from django.urls import reverse_lazy
 from django.utils.encoding import force_text
 from django.utils.translation import ugettext_lazy as _
 from django.views.generic import FormView, View
-from u2flib_server import u2f
+from fido2.server import Fido2Server, RelyingParty
 
-from .constants import (AUTHENTICATION_REQUEST_SESSION_KEY, AUTHENTICATION_USER_SESSION_KEY,
-                        REGISTRATION_REQUEST_SESSION_KEY, U2F_AUTHENTICATION_REQUEST, U2F_REGISTRATION_REQUEST)
-from .forms import U2fResponseForm
-from .models import U2fDevice
+from .constants import (AUTHENTICATION_USER_SESSION_KEY, FIDO2_AUTHENTICATION_REQUEST, FIDO2_REGISTRATION_REQUEST,
+                        FIDO2_REQUEST_SESSION_KEY)
+from .forms import Fido2AuthenticationForm, Fido2RegistrationForm
+from .models import Authenticator
 
 _LOGGER = logging.getLogger(__name__)
 
 
-class BaseU2fRequestView(View, metaclass=ABCMeta):
-    """Base view for U2F request views.
+class Fido2ViewMixin(object):
```

Comment:
I think there is a missing `get_attestation` method in one of the views for easy access to attestation object.

Reply:
I don't see a point in there.

Reply:
Attestation object can provide information needed to change the content of the page or view behavior, so it might be useful to have it accessible.

Reply:
Created #11

Reply:
Perfect!